### PR TITLE
Add advertise ip 

### PR DESCRIPTION
**Pull request by eloo**
Hi, 

this PR enables the emulated hue to use a custom ip for advertisement together with some fixed for the custom port configuration.

My main goal for this changes is to use the emulated hue in kubernetes together with metallb.

Addresses the following issue:
https://github.com/hass-emulated-hue/core/issues/513
https://github.com/hass-emulated-hue/core/issues/514

Example of my config using the "app-template" helm chart (https://github.com/bjw-s/helm-charts/tree/main/charts/other/app-template)

```
controllers:
  ...
  hass-emulated-hue:
    enabled: true
    pod:
      hostNetwork: true   # needed for upnp and zeroconf
      affinity: 
        nodeAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 1
            preference:
              matchExpressions:
              - key: node-role.kubernetes.io/control-plane
                operator: DoesNotExist
    containers:
      main:
        env: 
          HASS_URL: h****
          HTTP_PORT: 6547
          HTTPS_PORT: 6548
          VERBOSE: "true"
          ADVERTISE_IP: 192.168.11.97
          USE_DEFAULT_PORTS_FOR_DISCOVERY: "true"
          DATA_DIR: /config
        envFrom:
          - secretRef:
              name: emulated-hue-secrets
        image: 
          repository: docker.io/eloo/hass-emulated-hue
          tag: 0.4.0-dev
        probes:
          liveness:
            enabled: false
          readiness:
            enabled: false
          startup:
            enabled: false
service:
  ...
  emulated-hue:
    controller: hass-emulated-hue
    enabled: true
    type: LoadBalancer
    annotations:
      metallb.universe.tf/ip-allocated-from-pool: emulated-hue-pool
    ports:
      hue-http:
        port: 80
        targetPort: 6547
      hue-https:
        port: 443
        targetPort: 6548
persistence:
  ...
  hass-emulated-hue:
    enabled: true
    type: persistentVolumeClaim
    name: hass-emulated-hue
    accessMode: ReadWriteOnce
    size: 100Mi
    storageClass: nfs-client-ssd        
    advancedMounts:
      hass-emulated-hue:
        main:
          - path: /config

### CODE REVIEW

File: emulated_hue/__main__.py [hunk 1]
```diff
@@ -71,6 +77,7 @@
         action="store_true",
         help=f"Always use HTTP port {const.HUE_HTTP_PORT} and HTTPS port {const.HUE_HTTPS_PORT} for discovery "
         f"regardless of actual exposed ports. Useful with reverse proxy.",
+        default=os.getenv("USE_DEFAULT_PORTS_FOR_DISCOVERY", "false"),
```
Comment (alexyao2015) [inline emulated_hue/__main__.py]: Does the string "false" work here? Should this be `False` instead?

File: emulated_hue/controllers/config.py [hunk 2]
```diff
@@ -52,8 +53,13 @@ def __init__(
 
         # Get the IP address that will be passed to during discovery
         self._ip_addr = get_local_ip()
+        self._advertise_ip = self._ip_addr
         LOGGER.info("Auto detected listen IP address is %s", self.ip_addr)
 
+        if advertise_ip:
+            self._advertise_ip = advertise_ip
+            LOGGER.info("Use custom advertisement ip %s", self.advertise_ip)
```
Comment (alexyao2015) [inline emulated_hue/controllers/config.py]: Should be "Using" instead of "Use"

File: emulated_hue/controllers/models.py [hunk 3]
```diff
@@ -31,15 +31,15 @@ class EntityState(BaseModel):
...
-    transition_seconds: float | None = None
-    brightness: int | None = None
-    color_temp: int | None = None
-    hue_saturation: tuple[int, int] | None = None
-    xy_color: tuple[float, float] | None = None
-    rgb_color: tuple[int, int, int] | None = None
-    flash_state: str | None = None
-    effect: str | None = None
-    color_mode: str | None = None
+    transition_seconds: Optional[float] = None
```
Comment (alexyao2015) [inline emulated_hue/controllers/models.py]: Can you revert these type changes? These are the old style python type hints and shouldn't be used. More over, these changes should be in a separate PR.

File: emulated_hue/discovery.py [hunk 4]
```diff
@@ -39,13 +40,16 @@ def start_zeroconf_discovery(config: Config):
     info = ServiceInfo(
         zeroconf_type,
         name=f"Philips Hue - {config.bridge_id[-6:]}.{zeroconf_type}",
-        addresses=[get_ip_pton()],
+        addresses=[config.advertise_ip],
```
Comment (alexyao2015) [inline emulated_hue/discovery.py]: Looks like it should provide a socket as done with the original code instead of just a plain string.

**Review by alexyao2015 — CHANGES_REQUESTED** [review]
Thanks for the PR! I left a few small comments but overall it looks pretty good!

File: emulated_hue/__main__.py (hunk 1 above)
Comment (eloo) [reply emulated_hue/__main__.py]: nope its not working and also not necessary because its handled below manually.. 
but really bad that the used framework can not handle boolean env variables properly out of the box :(

File: emulated_hue/controllers/models.py (hunk 3 above)
Comment (eloo) [reply emulated_hue/controllers/models.py]: for sure
i have adopted this only from a different fork because the implementation in the main branch was not working at all

File: emulated_hue/discovery.py (hunk 4 above)
Comment (eloo) [reply emulated_hue/discovery.py]: good catch

fixed that

File: emulated_hue/controllers/models.py (hunk 3 above)
Comment (eloo) [reply emulated_hue/controllers/models.py]: i was able to fix the upstream issue by changing the tuple type of saturation to float instead of int

**Review by eloo — COMMENTED** [review]
@alexyao2015 thanks for the review
i hope i have addressed all your comments properly

File: emulated_hue/controllers/models.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 """Device state model."""
 import asyncio
 from dataclasses import dataclass
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Optional, Tuple
```
Comment (alexyao2015) [inline emulated_hue/controllers/models.py:4]: These imports are now no longer necessary.

**Review by alexyao2015 — CHANGES_REQUESTED** [review]
One more small change. Can you also ensure you run `pre-commit` locally? 

```
pip install pre-commit
pre-commit run --all-files
```

File: emulated_hue/controllers/models.py (hunk 3 above)
Comment (alexyao2015) [reply emulated_hue/controllers/models.py]: Interesting the master branch here isn't working. Can you explain a bit more regarding the reason for this change? I've been using the master branch for the past year and haven't experienced any issues.

**Comment by alexyao2015** [comment]
There should also be changed to the docker image to expose this configuration setting through bashio as well as in the hassio repo.

**Comment by eloo** [comment]
Hi,
as hass-emulated-hue seems to have still a lot of bugs i have switched to DIYHue which works pretty stable also in combination with Harmony Hub.

So i will close this PR as im no longer using this project. 

Feel free to use my work as draft for implementation for custom advertisements.
